torch/csrc/jit/frontend/schema_matching.cpp

```diff
@@ -284,6 +284,19 @@ static bool varargsCanBeUsedAsList(
       !typevar_list;
 }
 
+// Note (@zasdfgbnm):
+// This is a workaround for https://github.com/pytorch/pytorch/issues/47964
+// Currently JIT does not distinguish ScalarType vs int, so there is really
+// no way to distinguish x.view(1) vs x.view(torch.int8). So we have to hardcode
+// the aten::view.dtype here to block this overload. This blocklist should be
+// removed when JIT fully suports ScalarType as its own type.
+bool isBlockListedSchema(const FunctionSchema& schema) {
+  if (schema.name() == "aten::view" && schema.overload_name() == "dtype") {
+    return true;
+  }
+  return false;
+}
+
```

Comment:
I looked at the codegen, and some codes in dispatcher and JIT, and I feel that hard coding the operator name here is the best solution.

Reply:
It would be nice to allow it work with the unambiguous `torch.view(dtype=torch.int8)`

Reply:
Are you suggesting that if `schema.name() == "aten::view"` then I should check if `kwargs` is `"dtype"` and special case it to make this case work? I think this would require changing more places in `tryMatchSchema`. As a temporary workaround, I prefer the workaround to be simple, and can be easily reverted when the real fix (which is to add ScalarType support to JIT) is landed.